rdkit/Chem/UnitTestSuppliers.py

```diff
@@ -85,17 +85,19 @@ def test3SmilesSupplier(self):
 """
     RDLogger.DisableLog('rdApp.error')
 
-    fileN = tempfile.mktemp('.csv')
+    fileH, fileN = tempfile.mkstemp(suffix='.csv')
     try:
-      with open(fileN, 'w+') as f:
+      with os.fdopen(fileH, 'w') as f:
         f.write(txt)
+        f.close()
```

Comment:
Can you explain, why you needed to replace this? Do you need to do this and the `del suppl` for it to work?

```
from contextlib import closing
with contextlib.closing(os.fdopen(fileH, 'w')) as f:
          f.write(txt)
```